src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -1057,10 +1054,28 @@ void CascadiaSettings::WriteSettingsToDisk() const
                                               CREATE_NEW,
                                               FILE_ATTRIBUTE_NORMAL,
                                               nullptr) };
-    if (GetLastError() != ERROR_FILE_EXISTS)
+
+    // throw if backup file already exists
+    THROW_LAST_ERROR_IF(INVALID_HANDLE_VALUE == backupFile.get());
+    THROW_LAST_ERROR_IF(!WriteFile(backupFile.get(), content.data(), gsl::narrow<DWORD>(content.size()), nullptr, nullptr));

```

Comment:
@miniksa why exception handling? this is going to happen on every single save operation after the first, so it doesn't seem particularly exceptional

Reply:
RETURN_LAST_ERROR_IF and HRESULT it so it's not all exceptions? I dunno. I find it weird to make a backup ONCE and then never overwrite it again in the first place.

I feel like the backups should be versioned with datetime stamp or that the backup should be the most recent one, not the oldest one.

Reply:
My concern with creating a string of backups is that we'll be taking up more storage space on every save.

Reply:
Discussed offline. These backup files are _tiny_ (7kb). So we'll do a datetime stamp. Once our settings UI and model is good to go, we'll remove this (or make it configurable).